mcs/class/corlib/System.Reflection/CustomAttributeTypedArgument.cs

```diff
@@ -47,7 +49,7 @@ public struct CustomAttributeTypedArgument {
 				throw new ArgumentNullException ("argumentType");
 
 			this.argumentType = argumentType;
-			this.value = value;
+			this.value = (value == null) ? null : CanonicalizeValue (value);
```

Comment:
What about setting the value only when it's not null?

Reply:
In this case I get `error CS0171: Field 'CustomAttributeTypedArgument.value' must be fully assigned before control is returned to the caller`.

Reply:
oh, this is a struct (nevermind then)